Detectors/EMCAL/simulation/src/LZEROElectronics.cxx

```diff
@@ -0,0 +1,288 @@
+    // LOG(info) << "DIG SIMONE peakFinderOnPatch in LZEROElectronics: TimeSums[1] = " << TimeSums[1];
+    // LOG(info) << "DIG SIMONE peakFinderOnPatch in LZEROElectronics: TimeSums[2] = " << TimeSums[2];
+    // LOG(info) << "DIG SIMONE peakFinderOnPatch in LZEROElectronics: TimeSums[3] = " << TimeSums[3];
+  }
+  double integralOfADCvalues = 0;
+  for (auto it = TimeSums.begin(); it != TimeSums.end(); it++) {
+    integralOfADCvalues += *it;
+  }
+  // integralOfADCvalues += TimeSums[3];
+  // if( integralOfADCvalues != 0 )LOG(info) << "DIG SIMONE peakFinderOnPatch in LZEROElectronics: integralOfADCvalues = " << integralOfADCvalues;
+  bool peakOverThreshold = false;
+  if (integralOfADCvalues > mThreshold && trendOfDigitsInTower) {
+    peakOverThreshold = true;
+  }
+  return peakOverThreshold;
+}
+//_____________________________________________________________________
+// Peak finding algorithm on all patches
+// It fills the mPeakFound vector with potential 1s
+bool LZEROElectronics::peakFinderOnAllPatches(Patches& p)
+{
+  bool isFoundGlobal = false;
+  p.mFiredPatches.clear();
+  for (auto& patches : p.mIndexMapPatch) {
+    auto PatchID = std::get<0>(patches);
+    auto isFound = peakFinderOnPatch(p, PatchID);
+    if (isFound) {
+      p.mFiredPatches.push_back(PatchID);
+      isFoundGlobal = true;
+    }
+  }
+  return isFoundGlobal;
+}
+//________________________________________________________
+void LZEROElectronics::init()
+{
+  mSimParam = &(o2::emcal::SimParam::Instance());
+  mRandomGenerator = new TRandom3(std::chrono::high_resolution_clock::now().time_since_epoch().count());
+  mSimulateNoiseDigits = mSimParam->doSimulateNoiseDigits();
+  mTriggerMap = new TriggerMappingV2(mGeometry);
+  setThreshold(132.);
+  // setThreshold(66.);
+  // setThreshold(0.);
+}
+//________________________________________________________
+void LZEROElectronics::clear()
+{
+}
+//________________________________________________________
+void LZEROElectronics::updatePatchesADC(Patches& p)
+{
+  // LOG(info) << "DIG SIMONE updatePatchesADC in LZEROElectronics";
+  p.updateADC();
+}
+//_______________________________________________________________________
+void LZEROElectronics::addNoiseDigits(Digit& d1)
+{
+  double amplitude = d1.getAmplitude();
+  double sigmaHG = mSimParam->getPinNoise();
+  double sigmaLG = mSimParam->getPinNoiseLG();
+
+  uint16_t noiseHG = std::floor(std::abs(mRandomGenerator->Gaus(0, sigmaHG) / constants::EMCAL_TRU_ADCENERGY)); // ADC
+  // uint16_t noiseHG = std::floor(std::abs(mRandomGenerator->Gaus(0, sigmaHG) / constants::EMCAL_ADCENERGY)); // ADC
+  // uint16_t noiseLG = std::floor(std::abs(mRandomGenerator->Gaus(0, sigmaLG) / (constants::EMCAL_ADCENERGY * constants::EMCAL_HGLGFACTOR))); // ADC
+  uint16_t noiseLG = 0; // ADC
+
+  // MCLabel label(true, 1.0);
+  Digit d(d1.getTower(), noiseLG, noiseHG, d1.getTimeStamp());
+
+  d1 += d;
+}
```

Comment:
Ok Markus, I can move the method to the `DIgitizerTRU` when I create the digits to be passed to the `DigitsWriteoutBufferTRU`. I think I will keep the function essentially the same, just instead of passing a `noiseLG = 0` when I create the `Digit` I will explicitly pass 0 instead. Would this be agreeable with you? Is the definition of the `noiseHG` ok with you? I will of course rename it to `noise`, but I am afraid about the `sigmaHG`. Is it the same in the case of TRU?